gui/packages/desktop/src/main/index.js

```diff
@@ -157,24 +157,48 @@ const ApplicationMain = {
     }
   },
 
-  _onBeforeQuit() {
-    this._shouldQuit = true;
+  async _onBeforeQuit(event: Event) {
+    switch (this._quitStage) {
+      case 'unready':
+        // postpone the app shutdown
+        event.preventDefault();
 
-    if (process.env.NODE_ENV === 'development') {
```

Comment:
FYI 

Previously the call to `app.quit()` happened in response to `window-all-closed` event only so it was necessary to kill all windows in order to get the app to quit.

Closing dev tools is not necessary anymore because when the app moves to "ready to quit" stage it calls `app.quit()` automatically.